src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -439,10 +439,51 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
             {
                 case 1:
                     assert(!instrIsRMW);
+
                     if (targetReg != falseReg)
                     {
-                        GetEmitter()->emitIns_R_R(INS_sve_movprfx, EA_SCALABLE, targetReg, falseReg);
+                        // If targetReg is not the same as `falseReg` then need to move
+                        // the `falseReg` to `targetReg`.
+
+                        if (intrin.op1->IsMaskAllBitsSet() && intrin.op3->IsVectorZero())
+                        {
+                            // We already skip importing ConditionalSelect if op1 == trueAll, however
+                            // if we still see it here, it is because we wrapped the predicated instruction
+                            // inside ConditionalSelect.
+                            // As such, no need to move the `falseReg` to `targetReg`
+                            // because the predicated instruction will eventually set it.
+                            assert(falseReg == REG_NA);
```

Comment:
This could actually just be combined with the next branch. That is rather than having:
```cpp
if (intrin.op1->IsMaskAllBitsSet() && intrin.op3->IsVectorZero())
{
    assert(falseReg == REG_NA);
}
else if (intrin.op3->isContained())
{
    assert(falseReg == REG_NA);
    assert(intrin.op3->IsVectorZero());
    GetEmitter()->emitIns_R_R_R(INS_sve_movprfx, emitSize, targetReg, maskReg, targetReg, opt);
}
```

you could just have:
```cpp
if (intrin.op3->isContained())
{
    assert(intrin.op3->IsVectorZero());

    if (intrin.op1->isContained())
    {
        assert(intrin.op1->IsMaskAllBitsSet());
    }
    else
    {
        GetEmitter()->emitIns_R_R_R(INS_sve_movprfx, emitSize, targetReg, maskReg, targetReg, opt);
    }
}
```